add-on/src/popup/browser-action/store.js

```diff
@@ -204,15 +211,17 @@ export default (state, emitter) => {
   emitter.on('toggleActive', async () => {
     const prev = state.active
     state.active = !prev
-    if (!state.active) {
-      state.gatewayAddress = state.pubGwURLString
-      state.ipfsApiUrl = null
-      state.gatewayVersion = null
-      state.swarmPeers = null
-      state.isIpfsOnline = false
-    }
     try {
       await browser.storage.local.set({ active: state.active })
+      if (!state.active) {
+        state.gatewayAddress = state.pubGwURLString
+        state.ipfsApiUrl = null
+        state.gatewayVersion = null
+        state.swarmPeers = null
+        state.isIpfsOnline = false
+      }
+      await notifyOptionChange()
+      await browser.storage.local.set({ active: state.active })
```

Comment:
Why aren't we persisting the rest of the state?
```suggestion
      await browser.storage.local.set({ ...state })
```